wrangler-service/src/test/java/co/cask/wrangler/service/bigquery/BigQueryServiceTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.service.bigquery;
+
+import co.cask.wrangler.dataset.connections.Connection;
+import co.cask.wrangler.service.connections.ConnectionType;
+import co.cask.wrangler.service.gcp.GCPUtils;
+import com.google.cloud.bigquery.DatasetId;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Tests for {@link BigQueryService}.
+ */
+public class BigQueryServiceTest {
+
+  @Test
+  public void testDatasetWhitelistParsing() {
+    Connection connection = new Connection();
+    connection.setType(ConnectionType.BIGQUERY);
+    connection.setName("test");
+    connection.setId("test");
+    connection.putProp(GCPUtils.PROJECT_ID, "pX");
+    connection.putProp("datasetWhitelist", "p0:d0,p1:d1,d2,:d3");
```

Comment:
Can user just put project? i.e p3 or p3:

Reply:
no. 'p3:' is checked and ignored (can add that to the unit test). 'p3' would be interpreted as dataset 'p3' in the connection project.

Reply:
👍